flink-end-to-end-tests/test-scripts/test_high_parallelism_iterations.sh

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env bash
+################################################################################
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+################################################################################
+
+source "$(dirname "$0")"/common.sh
+
+PARALLELISM="${PARALLELISM:-100}"
+
+TEST=flink-high-parallelism-iterations-test
+TEST_PROGRAM_NAME=HighParallelismIterationsTestProgram
+TEST_PROGRAM_JAR=$TEST_INFRA_DIR/../../flink-end-to-end-tests/$TEST/target/$TEST_PROGRAM_NAME.jar
+
+echo "Run Not So MiniCluster Iterations Graph Connected Components Program"
+
+cp $FLINK_DIR/conf/flink-conf.yaml $FLINK_DIR/conf/flink-conf.yaml.bak
```

Comment:
I think your job upgrade e2e test is making the backup and restore of the configuration a utility function. I suggest that we also use the function from common here once it is merged to avoid code duplication.